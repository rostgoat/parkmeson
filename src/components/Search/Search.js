import React, { useState } from 'react'

import { SearchBar } from 'react-native-elements'

const Search = () => {
  const [search, updateSearch] = useState()

  return (
    <SearchBar
      placeholder="Where do you need parking?"
      onChangeText={updateSearch}
      value={search}
    />
  )
}

export default Search
