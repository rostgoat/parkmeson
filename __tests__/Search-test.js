/**
 * @format
 */

import 'react-native'
import React from 'react'

import renderer from 'react-test-renderer'

import { Search } from '../src/components/Search'

// Note: test renderer must be required after react-native.

it('renders correctly', () => {
  const tree = renderer.create(<Search />).toJSON()
  expect(tree).toMatchSnapshot()
})
