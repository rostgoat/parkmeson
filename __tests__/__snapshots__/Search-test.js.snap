// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#393e42",
      "borderBottomColor": "#000",
      "borderBottomWidth": 1,
      "borderTopColor": "#000",
      "borderTopWidth": 1,
      "padding": 8,
    }
  }
>
  <View
    style={
      Object {
        "paddingHorizontal": 0,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#303337",
          "borderBottomWidth": 0,
          "borderColor": "#86939e",
          "borderRadius": 3,
          "flexDirection": "row",
          "minHeight": 30,
          "overflow": "hidden",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 8,
            "marginVertical": 4,
            "paddingRight": 4,
          }
        }
      >
        <View
          style={
            Object {
              "overflow": "hidden",
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#86939e",
                      "fontSize": 18,
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="iconIcon"
              >
                î¢¶
              </Text>
            </View>
          </View>
        </View>
      </View>
      <TextInput
        allowFontScaling={true}
        editable={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        onSubmitEditing={[Function]}
        placeholder="Where do you need parking?"
        placeholderTextColor="#86939e"
        platform="default"
        rejectResponderTermination={true}
        style={
          Object {
            "alignSelf": "center",
            "color": "#86939e",
            "flex": 1,
            "fontSize": 18,
            "marginLeft": 10,
            "minHeight": 40,
          }
        }
        testID="searchInput"
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "marginRight": 8,
            "marginVertical": 4,
            "paddingRight": 4,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        />
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "#ff190c",
          "fontSize": 12,
          "height": 0,
          "margin": 0,
          "padding": 0,
        }
      }
    />
  </View>
</View>
`;
